up-l3/usubscription/v3/README.adoc

```diff
@@ -18,139 +18,619 @@ SPDX-FileType: DOCUMENTATION
+
+The `uSubscription` service component provides this bookkeeping capability.  
+
+[#usubscription-publisher]
+== Publishers
+
+Publishers make information available to the uProtocol network, typically by using the mechanisms provided by the uProtocol language library and up-transport implementation appropriate for their context. 
+
+A Publisher in a uProtocol network ususally does not perform any additional actions beyond what is necessary to publish messages to the up-transport it is using.
+
+[#usubscription-subscriber]
+== Subscribers
+
+Subscribers receive information from the uProtocol network, typically by using the mechanisms provided by the uProtocol language library and up-transport implementation appropriate for their context. 
+
+[.specitem,oft-sid="dsn~usubscribe-interaction-subscriber~1"]
+--
+In addition to performing the actions necessary to subscribe to and receive messages from the up-transport that they are using, Publishers in a uProtocol network *MUST* interact with their local uSubscription service instance, especially to inform about what messages they `subscribe()` to or `unsubscribe()` from.
+--
+
+=== Subscribe()
+
+[.specitem,oft-sid="req~usubscribe-subscribe-protobuf~1",oft-needs="impl"]
+--
+The uSubscription service `Subscribe()` function *MUST* implement the corresponding protobuf definition in link:../up-core-api/uprotocol/core/uSubscription/v3/usubscription.proto[usubscription.proto].
+--
 
+[.specitem,oft-sid="req~usubscribe-subscribe~1",oft-needs="impl,utest"]
+--
+When a client calls the `Subscribe()` function, uSubscription service *MUST* store the client as a subscriber of the provided topic for the duration of the uSubscription service runtime (no persistency across service restarts).
+--
 
-NOTE: uSubscription interface is declared in link:../../../up-core-api/uprotocol/core/usubscription/v3/usubscription.proto[usubscription.proto] and follows link:../../../basics/error_model.adoc[uProtocol Error Model] for return codes.
+[.specitem,oft-sid="req~usubscribe-subscribe-multiple~1",oft-needs="impl,utest"]
+--
+When a client calls the `Subscribe()` function for a topic that it is already registered for, uSubscription service *SHALL* return a response message containing the current subscription status for this client-topic combination (e.g. `SUBSCRIBED` or `SUBSCRIBE_PENDING`).
+--
 
+[.specitem,oft-sid="req~usubscribe-subscribe-notifications~1",oft-needs="impl,utest"]
+--
+When a client calls the `Subscribe()` function, uSubscription service *MUST* generate subscription change notifications that indicate any change of subscription state of the subscribed topic.
```

Comment:
FMPOV this does not belong here but instead should be specified as part of the `RegisterForNotifications` operation.
These notifications are only sent **if** the client indeed registers a corresponding listener, or am I mistaken?

Reply:
As far as I understand the original spec, the general/default notification channel should always get subscription change notifications. RegisterForNotifications is for setting a custom, client-specific topic/channel to receive these notifications on.

Not saying that I am unwilling to change any of this, but this is how I read the original intent.

Reply:
You are the expert :-)

Reply:
Only relatively speaking ;-) - but let's keep this on the backburner for potential future evolution.